src/core/editor.coffee

```diff
@@ -167,12 +173,37 @@ class Editor
       )
     )
 
+  _splitDelta: (delta, point) ->
+    return [
+      delta.slice(0, point),
+      delta.slice(point),
+    ]
+
+  _mergeDelta: (deltaA, deltaB) ->
+    return new Delta(deltaA.ops.concat(deltaB.ops))
+
   _trackDelta: (fn) ->
+    oldIndex = @savedRange?.start
     fn()
     newDelta = @doc.toDelta()
-    # TODO need to get this to prefer earlier insertions
-    delta = @delta.diff(newDelta)
-    return delta
+    diff = null
+    newIndex = @selection.getRange()?.start
+    try
+      if oldIndex? and newIndex? and oldIndex <= @delta.length() and newIndex <= newDelta.length()
+        [oldLeftDelta, oldRightDelta] = this._splitDelta(@delta, oldIndex)
+        [newLeftDelta, newRightDelta] = this._splitDelta(newDelta, newIndex)
+        diffLeft = oldLeftDelta.diff(newLeftDelta)
+        diffRight = oldRightDelta.diff(newRightDelta)
+        diff = this._mergeDelta(diffLeft, diffRight)
+    catch ignored
+
+    if @selection.getRange()
```

Comment:
It'd be more optimal to assign the result of `getRange()` to a temporary variable since the call is not free